visualizer/examples.js

```diff
@@ -6,11 +6,31 @@
   if (typeof exports === 'object') {
     module.exports = initModule;
   } else {
-    initModule(root.ohm, root.ohmEditor, root.domUtil);
+    initModule(root.ohm, root.ohmEditor, root.domUtil, root.CheckedEmitter);
   }
-})(this, function(ohm, ohmEditor, domUtil) {
+})(this, function(ohm, ohmEditor, domUtil, CheckedEmitter) {
   var idCounter = 0;
   var selectedId = -1;
+  var exampleValues = Object.create(null);
+
+  // Exports
+  // -------
+
+  ohmEditor.examples = Object.assign(new CheckedEmitter(), {
+    addExample: addExample,
+    getExample: getExample,
+    getExamples: getExamples,
+    setExample: setExample,
+    setSelected: setSelected,
+    saveExamples: saveExamples
+  });
+
+  ohmEditor.examples.registerEvents({
+    'add:example': ['id'],
+    'set:example': ['id', 'oldValue', 'newValue'],
+    'set:selected': ['id'],
+    'remove:example': ['id']
```

Comment:
Can you put a comment above each line talking about when the event is thrown?
In the existing system, some events fire before, some after the action is/was processed.